### PR TITLE
The 3rd batch patch for msdn url

### PR DESCRIPTION
**Pull request by luyajun0205**
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Comment by luyajun0205** [comment]
@mairaw @BillWagner @TianqiZhang This is the 2nd Patch, please help to review it.

File: xml/System.ServiceModel.Configuration/NetNamedPipeBindingCollectionElement.xml [hunk 1]
```diff
@@ -21,7 +21,7 @@
   <Docs>
     <summary>Represents a configuration section that contains a collection of <see cref="T:System.ServiceModel.Configuration.NetNamedPipeBindingElement" /> instances.</summary>
     <remarks>To be added.</remarks>
-    <related type="Article" href="https://msdn.microsoft.com/library/00a8580b-face-47a4-838d-b9fed48e72df">netNamedPipeBinding Element</related>
+    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/netnamedpipebinding">netNamedPipeBinding Element</related>
```
Comment (mairaw) [inline xml/System.ServiceModel.Configuration/NetNamedPipeBindingCollectionElement.xml:24]:
```suggestion
    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/netnamedpipebinding">&lt;netNamedPipeBinding&gt;</related>
```

File: xml/System.ServiceModel.Configuration/NetNamedPipeBindingElement.xml [hunk 2]
```diff
@@ -25,7 +25,7 @@
  ]]></format>
     </remarks>
     <altmember cref="T:System.ServiceModel.NetNamedPipeBinding" />
-    <related type="Article" href="https://msdn.microsoft.com/library/00a8580b-face-47a4-838d-b9fed48e72df">netNamedPipeBinding Element</related>
+    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/netnamedpipebinding">netNamedPipeBinding Element</related>
```
Comment (mairaw) [inline xml/System.ServiceModel.Configuration/NetNamedPipeBindingElement.xml:28]:
```suggestion
    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/netnamedpipebinding">&lt;netNamedPipeBinding&gt;</related>
```

File: xml/System.ServiceModel.Configuration/NetTcpBindingCollectionElement.xml [hunk 3]
```diff
@@ -22,7 +22,7 @@
     <summary>Represents a configuration element that contains a collection of <see cref="T:System.ServiceModel.Configuration.NetTcpBindingElement" /> instances.</summary>
     <remarks>To be added.</remarks>
     <altmember cref="T:System.ServiceModel.NetTcpBinding" />
-    <related type="Article" href="https://msdn.microsoft.com/library/5c5104a7-8754-4335-8233-46a45322503e">netTcpBinding Element</related>
+    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/nettcpbinding">netTcpBinding Element</related>
```
Comment (mairaw) [inline xml/System.ServiceModel.Configuration/NetTcpBindingCollectionElement.xml:25]:
```suggestion
    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/nettcpbinding">&lt;netTcpBinding&gt;</related>
```

File: xml/System.ServiceModel.Configuration/ReliableSessionElement.xml [hunk 4]
```diff
@@ -25,7 +25,7 @@
  ]]></format>
     </remarks>
     <altmember cref="T:System.ServiceModel.ReliableSession" />
-    <related type="Article" href="https://msdn.microsoft.com/library/129b4a59-37f0-4030-b664-03795d257d29">reliableSession element</related>
+    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/reliablesession">reliableSession element</related>
```
Comment (mairaw) [inline xml/System.ServiceModel.Configuration/ReliableSessionElement.xml:28]:
```suggestion
    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/reliablesession">&lt;reliableSession&gt;</related>
```

File: xml/System.ServiceModel.Configuration/SecurityElement.xml [hunk 5]
```diff
@@ -17,7 +17,7 @@
   <Docs>
     <summary>Represents a configuration element that specifies the security options for a custom binding. This class cannot be inherited.</summary>
     <remarks>To be added.</remarks>
-    <related type="Article" href="https://msdn.microsoft.com/library/9da4f960-f64e-4d8a-894d-2b09eba5ce4b">customBinding Element</related>
+    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/custombinding">customBinding Element</related>
```
Comment (mairaw) [inline xml/System.ServiceModel.Configuration/SecurityElement.xml:20]:
```suggestion
    <related type="Article" href="/dotnet/framework/configure-apps/file-schema/wcf/custombinding">&lt;customBinding&gt;</related>
```

File: xml/System.ServiceModel/ServiceSecurityContext.xml [hunk 6]
```diff
@@ -62,10 +62,10 @@
     <altmember cref="T:System.IdentityModel.Claims.Claim" />
     <altmember cref="T:System.IdentityModel.Claims.ClaimSet" />
     <altmember cref="T:System.ServiceModel.ServiceAuthorizationManager" />
-    <related type="Article" href="https://msdn.microsoft.com/library/389b5a57-4175-4bc0-ada0-fc750d51149f">How To: Examine the Security Context</related>
-    <related type="Article" href="https://msdn.microsoft.com/library/1db325ec-85be-47d0-8b6e-3ba2fdf3dda0">Authorization Policy Sample</related>
-    <related type="Article" href="https://msdn.microsoft.com/library/f0ecc6f7-f4b5-42a4-9cb1-b02e28e26620">Securing Services</related>
-    <related type="Article" href="https://msdn.microsoft.com/library/099defbb-5d35-434e-9336-1a49b9ec7663">Claims and Authorization</related>
+    <related type="Article" href="/dotnet/framework/wcf/how-to-examine-the-security-context">How To: Examine the Security Context</related>
+    <related type="Article" href="/dotnet/framework/wcf/samples/authorization-policy">Authorization Policy Sample</related>
```
Comment (mairaw) [inline xml/System.ServiceModel/ServiceSecurityContext.xml:66]:
```suggestion
    <related type="Article" href="/dotnet/framework/wcf/samples/authorization-policy">Authorization Policy</related>
```

File: xml/System.ServiceModel/ServiceSecurityContext.xml [hunk 7]
```diff
@@ -62,10 +62,10 @@
...
+    <related type="Article" href="/dotnet/framework/wcf/securing-services">Securing Services</related>
+    <related type="Article" href="/dotnet/framework/wcf/feature-details/managing-claims-and-authorization-with-the-identity-model">Claims and Authorization</related>
```
Comment (mairaw) [inline xml/System.ServiceModel/ServiceSecurityContext.xml:68]:
```suggestion
    <related type="Article" href="/dotnet/framework/wcf/feature-details/managing-claims-and-authorization-with-the-identity-model">Managing Claims and Authorization with the Identity Model</related>
```

File: xml/System.ServiceModel/XmlSerializerFormatAttribute.xml [hunk 8]
```diff
@@ -74,7 +74,7 @@
     <altmember cref="T:System.ServiceModel.Description.XmlSerializerOperationBehavior" />
     <altmember cref="T:System.Runtime.Serialization.DataContractSerializer" />
     <altmember cref="T:System.Xml.Serialization.XmlSerializer" />
-    <related type="Article" href="https://msdn.microsoft.com/library/c680602d-39d3-44f1-bf22-8e6654ad5069">Using the XmlSerializer</related>
+    <related type="Article" href="/dotnet/framework/wcf/feature-details/using-the-xmlserializer-class">Using the XmlSerializer</related>
```
Comment (mairaw) [inline xml/System.ServiceModel/XmlSerializerFormatAttribute.xml:77]:
```suggestion
    <related type="Article" href="/dotnet/framework/wcf/feature-details/using-the-xmlserializer-class">Using the XmlSerializer Class</related>
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 9]
```diff
@@ -498,7 +498,7 @@
 <Advertisements>  
   <Ad>  
     <ImageUrl>~/Images/ad1.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com/vbasic/</NavigateUrl>  
+    <NavigateUrl>https://visualstudio.microsoft.com/</NavigateUrl>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:501]:
```suggestion
    <NavigateUrl>https://docs.microsoft.com/dotnet/visual-basic/</NavigateUrl>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 10]
```diff
@@ -498,7 +498,7 @@
 <Advertisements>  
   <Ad>  
     <ImageUrl>~/Images/ad1.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com/vbasic/</NavigateUrl>  
+    <NavigateUrl>https://visualstudio.microsoft.com/</NavigateUrl>  
     <AlternateText>Visual Basic Developer Site</AlternateText>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:502]:
```suggestion
    <AlternateText>Visual Basic documentation</AlternateText>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 11]
```diff
@@ -510,7 +510,7 @@
   </Ad>  
   <Ad>  
     <ImageUrl>~/Images/ad2.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com</NavigateUrl>  
+    <NavigateUrl>https://docs.microsoft.com/en-us/</NavigateUrl>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:513]:
```suggestion
    <NavigateUrl>https://docs.microsoft.com/</NavigateUrl>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 12]
```diff
@@ -510,7 +510,7 @@
   </Ad>  
   <Ad>  
     <ImageUrl>~/Images/ad2.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com</NavigateUrl>  
+    <NavigateUrl>https://docs.microsoft.com/en-us/</NavigateUrl>  
     <AlternateText>MSDN</AlternateText>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:514]:
```suggestion
    <AlternateText>Docs</AlternateText>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 13]
```diff
@@ -629,7 +629,7 @@
 <Advertisements>  
   <Ad>  
     <ImageUrl>~/Images/ad1.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com/vbasic/</NavigateUrl>  
+    <NavigateUrl>https://visualstudio.microsoft.com/</NavigateUrl>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:632]:
```suggestion
    <NavigateUrl>https://docs.microsoft.com/dotnet/visual-basic/</NavigateUrl>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 14]
```diff
@@ -629,7 +629,7 @@
 <Advertisements>  
   <Ad>  
     <ImageUrl>~/Images/ad1.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com/vbasic/</NavigateUrl>  
+    <NavigateUrl>https://visualstudio.microsoft.com/</NavigateUrl>  
     <AlternateText>Visual Basic Developer Site</AlternateText>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:633]:
```suggestion
    <AlternateText>Visual Basic documentation</AlternateText>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 15]
```diff
@@ -642,7 +642,7 @@
   </Ad>  
   <Ad>  
     <ImageUrl>~/Images/ad2.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com</NavigateUrl>  
+    <NavigateUrl>https://docs.microsoft.com/en-us/</NavigateUrl>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:645]:
```suggestion
    <NavigateUrl>https://docs.microsoft.com/</NavigateUrl>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 16]
```diff
@@ -642,7 +642,7 @@
   </Ad>  
   <Ad>  
     <ImageUrl>~/Images/ad2.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com</NavigateUrl>  
+    <NavigateUrl>https://docs.microsoft.com/en-us/</NavigateUrl>  
     <AlternateText>MSDN</AlternateText>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:646]:
```suggestion
    <AlternateText>Docs</AlternateText>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 17]
```diff
@@ -787,7 +787,7 @@
 <Advertisements>  
   <Ad>  
     <ImageUrl>~/Images/ad1.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com/vbasic/</NavigateUrl>  
+    <NavigateUrl>https://visualstudio.microsoft.com/</NavigateUrl>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:790]:
```suggestion
    <NavigateUrl>https://docs.microsoft.com/dotnet/visual-basic/</NavigateUrl>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 18]
```diff
@@ -787,7 +787,7 @@
 <Advertisements>  
   <Ad>  
     <ImageUrl>~/Images/ad1.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com/vbasic/</NavigateUrl>  
+    <NavigateUrl>https://visualstudio.microsoft.com/</NavigateUrl>  
     <AlternateText>Visual Basic Developer Site</AlternateText>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:791]:
```suggestion
    <AlternateText>Visual Basic documentation</AlternateText>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 19]
```diff
@@ -799,7 +799,7 @@
   </Ad>  
   <Ad>  
     <ImageUrl>~/Images/ad2.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com</NavigateUrl>  
+    <NavigateUrl>https://docs.microsoft.com/en-us/</NavigateUrl>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:802]:
```suggestion
    <NavigateUrl>https://docs.microsoft.com/</NavigateUrl>  
```

File: xml/System.Web.UI.WebControls/AdRotator.xml [hunk 20]
```diff
@@ -799,7 +799,7 @@
   </Ad>  
   <Ad>  
     <ImageUrl>~/Images/ad2.gif</ImageUrl>  
-    <NavigateUrl>https://msdn.microsoft.com</NavigateUrl>  
+    <NavigateUrl>https://docs.microsoft.com/en-us/</NavigateUrl>  
     <AlternateText>MSDN</AlternateText>
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AdRotator.xml:803]:
```suggestion
    <AlternateText>Docs</AlternateText>  
```

File: xml/System.Web.UI.WebControls/AutoCompleteType.xml [hunk 21]
```diff
@@ -30,7 +30,7 @@
 
  Refer to your browser documentation for details on configuring and enabling the AutoComplete feature. For example, to enable the AutoComplete feature in Internet Explorer version 5 or later, from the **Tools** menu, choose **Internet Options**. Then choose the **Content** tab. Choose the **AutoComplete** button to view and modify options for the AutoComplete feature.
 
- For more information on the AutoComplete feature in Internet Explorer, search for [Using AutoComplete in HTML Forms](https://msdn.microsoft.com/library/default.asp?url=/workshop/author/forms/autocomplete_ovr.asp) in the MSDN Library at https://msdn.microsoft.com/library.
+ For more information on the AutoComplete feature in Internet Explorer, search for [Using AutoComplete in HTML Forms](https://docs.microsoft.com/en-us/) in the MSDN Library at https://docs.microsoft.com/en-us/.
```
Comment (mairaw) [inline xml/System.Web.UI.WebControls/AutoCompleteType.xml:35]:
```suggestion
```

File: xml/System.Windows.Forms/Form.xml [hunk 22]
```diff
@@ -1694,7 +1694,7 @@
   
  You can use this property to position your form relative to other forms and applications on the Windows desktop.  
   
- If you call the <xref:System.Windows.Forms.Form.SetDesktopLocation%2A> method before calling the <xref:System.Windows.Forms.Form.Show%2A> method, your form will be positioned at its default location, which is determined by the operating system. For more information about window positioning, see the "Window Size and Position" section of the "Window Features" document in the MSDN library at https://msdn.microsoft.com/library.  
+ If you call the <xref:System.Windows.Forms.Form.SetDesktopLocation%2A> method before calling the <xref:System.Windows.Forms.Form.Show%2A> method, your form will be positioned at its default location, which is determined by the operating system. For more information about window positioning, see the "Window Size and Position" section of the "Window Features" document in the MSDN library at https://docs.microsoft.com/en-us/.
```
Comment (mairaw) [inline xml/System.Windows.Forms/Form.xml:1697]:
```suggestion
 If you call the <xref:System.Windows.Forms.Form.SetDesktopLocation%2A> method before calling the <xref:System.Windows.Forms.Form.Show%2A> method, your form will be positioned at its default location, which is determined by the operating system. For more information about window positioning, see the [Window Size and Position](/windows/win32/winmsg/window-features#window-size-and-position) section of the "Window Features" article.  
```

File: xml/System.Windows.Forms/HtmlDocument.xml [hunk 23]
```diff
@@ -496,7 +496,7 @@
  ]]></format>
         </remarks>
         <related type="Article" href="https://msdn.microsoft.com/library/ms537500.aspx">Character Set Recognition</related>
```
Comment (mairaw) [inline xml/System.Windows.Forms/HtmlDocument.xml:498]:
```suggestion
```

File: xml/System.Windows.Forms/HtmlDocument.xml [hunk 24]
```diff
@@ -563,7 +563,7 @@
         <exception cref="T:System.ArgumentException">The argument for the <c>Domain</c> property must be a valid domain name using Domain Name System (DNS) conventions.</exception>
         <related type="Article" href="https://msdn.microsoft.com/library/aa752575.aspx">domain Property</related>
```
Comment (mairaw) [inline xml/System.Windows.Forms/HtmlDocument.xml:564]:
```suggestion
        <related type="Article" href="https://docs.microsoft.com/previous-versions/windows/internet-explorer/ie-developer/platform-apis/aa752575(v=vs.85)">domain Property</related>
```

File: xml/System.Windows.Forms/HtmlDocument.xml [hunk 25]
```diff
@@ -563,7 +563,7 @@
...
         <related type="Article" href="https://msdn.microsoft.com/library/ms533028.aspx">About Cross-Frame Scripting and Security</related>
```
Comment (mairaw) [inline xml/System.Windows.Forms/HtmlDocument.xml:565]:
```suggestion
```

File: xml/System.Windows.Forms/HtmlElement.xml [hunk 26]
```diff
@@ -1846,7 +1846,7 @@
   
  ]]></format>
         </remarks>
-        <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms534302.aspx">offsetParent Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/htmlelement/offsetparent">offsetParent Property</related>
         <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms530302.aspx">Measuring Element Dimension and Location</related>
```
Comment (mairaw) [inline xml/System.Windows.Forms/HtmlElement.xml:1850]:
```suggestion
```

File: xml/System.Windows.Forms/HtmlElement.xml [hunk 27]
```diff
@@ -1919,10 +1919,10 @@
...
-        <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms534303.aspx">offsetTop Property</related>
-        <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms534304.aspx">offsetWidth Property</related>
-        <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms534199.aspx">offsetHeight Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/htmlelement/offsetleft">offsetLeft Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/htmlelement/offsettop">offsetTop Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/htmlelement/offsetwidth">offsetWidth Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/htmlelement/offsetheight">offsetHeight Property</related>
         <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms530302.aspx">Measuring Element Dimension and Location</related>
```
Comment (mairaw) [inline xml/System.Windows.Forms/HtmlElement.xml:1926]:
```suggestion
```

File: xml/System.Windows.Forms/HtmlElement.xml [hunk 28]
```diff
@@ -2319,10 +2319,10 @@
...
-        <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms534618.aspx">scrollTop Property</related>
-        <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms534615.aspx">scrollHeight Property</related>
-        <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms534619.aspx">scrollWidth Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/element/scrollleft">scrollLeft Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/element/scrolltop">scrollTop Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/element/scrollheight">scrollHeight Property</related>
+        <related type="ExternalDocumentation" href="https://developer.mozilla.org/en-us/docs/web/api/element/scrollwidth">scrollWidth Property</related>
         <related type="ExternalDocumentation" href="https://msdn.microsoft.com/library/ms534312.aspx">overflow Property</related>
```
Comment (mairaw) [inline xml/System.Windows.Forms/HtmlElement.xml:2326]:
```suggestion
```

File: xml/System.Windows.Forms/MenuStrip.xml [hunk 29]
```diff
@@ -677,7 +677,7 @@ The <xref:System.Windows.Forms.MenuStrip> replaces and extends the <xref:System.
...
+ The following code example demonstrates overriding the <xref:System.Windows.Forms.Control.WndProc%2A> method to handle operating system messages identified in the <xref:System.Windows.Forms.Message> structure. The WM_ACTIVATEAPP operating system message is handled in this example to know when another application is becoming active. Refer to the Platform SDK documentation reference located in the MSDN library at https://docs.microsoft.com/en-us/ to understand the available <xref:System.Windows.Forms.Message.Msg%2A?displayProperty=nameWithType>, <xref:System.Windows.Forms.Message.LParam%2A?displayProperty=nameWithType>, and <xref:System.Windows.Forms.Message.WParam%2A?displayProperty=nameWithType> values. Actual constant values can be found in the windows.h header file included in the Platform SDK (Core SDK section) download, which is also available on MSDN.
```
Comment (mairaw) [inline xml/System.Windows.Forms/MenuStrip.xml:680]:
```suggestion
 The following code example demonstrates overriding the <xref:System.Windows.Forms.Control.WndProc%2A> method to handle operating system messages identified in the <xref:System.Windows.Forms.Message> structure. The WM_ACTIVATEAPP operating system message is handled in this example to know when another application is becoming active. Refer to the Platform SDK documentation reference located at <https://docs.microsoft.com> to understand the available <xref:System.Windows.Forms.Message.Msg%2A?displayProperty=nameWithType>, <xref:System.Windows.Forms.Message.LParam%2A?displayProperty=nameWithType>, and <xref:System.Windows.Forms.Message.WParam%2A?displayProperty=nameWithType> values. Actual constant values can be found in the windows.h header file included in the Platform SDK (Core SDK section) download, which is also available on Docs.  
```

File: xml/System.Windows.Forms/OSFeature.xml [hunk 30]
```diff
@@ -247,7 +247,7 @@
           <format type="text/markdown"><![CDATA[  
   
 ## Remarks  
- A layered window can be made transparent or translucent by the operating system. Support for layered windows was first introduced in Windows 2000. For more information, see the "Layered Windows: A New Way to Use Translucency and Transparency Effects in Windows Applications" in the Platform SDK documentation at https://msdn.microsoft.com.  
+ A layered window can be made transparent or translucent by the operating system. Support for layered windows was first introduced in Windows 2000. For more information, see the "Layered Windows: A New Way to Use Translucency and Transparency Effects in Windows Applications" in the Platform SDK documentation at https://docs.microsoft.com/en-us/.
```
Comment (mairaw) [inline xml/System.Windows.Forms/OSFeature.xml:250]:
```suggestion

A layered window can be made transparent or translucent by the operating system. Support for layered windows was first introduced in Windows 2000.
```

File: xml/System.Windows.Forms/SplitContainer.xml [hunk 31]
```diff
@@ -2648,7 +2648,7 @@
 ## Remarks  
  All messages are sent to the <xref:System.Windows.Forms.Control.WndProc%2A> method after getting filtered through the <xref:System.Windows.Forms.Control.PreProcessMessage%2A> method.  
   
- The <xref:System.Windows.Forms.Control.WndProc%2A> method corresponds exactly to the Windows `WindowProc` function. For more information about processing Windows messages, see the `WindowProc` function in the Platform SDK documentation at https://msdn.microsoft.com.  
+ The <xref:System.Windows.Forms.Control.WndProc%2A> method corresponds exactly to the Windows `WindowProc` function. For more information about processing Windows messages, see the `WindowProc` function in the Platform SDK documentation at https://docs.microsoft.com/en-us/.
```
Comment (mairaw) [inline xml/System.Windows.Forms/SplitContainer.xml:2651]:
```suggestion
 The <xref:System.Windows.Forms.Control.WndProc%2A> method corresponds exactly to the Windows `WindowProc` function. For more information about processing Windows messages, see the [WindowProc function](https://docs.microsoft.com/previous-versions/windows/desktop/legacy/ms633573(v=vs.85)).  
```

File: xml/System.Windows.Forms/ToolStrip.xml [hunk 32]
```diff
@@ -4670,10 +4670,10 @@
 ## Remarks  
  All messages are sent to the <xref:System.Windows.Forms.Control.WndProc%2A> method after getting filtered through the <xref:System.Windows.Forms.Control.PreProcessMessage%2A> method.  
   
- The <xref:System.Windows.Forms.Control.WndProc%2A> method corresponds exactly to the Windows `WindowProc` function. For more information about processing Windows messages, see the `WindowProc` function documentation in the Windows Platform SDK reference located in the MSDN library at https://msdn.microsoft.com/library.
+ The <xref:System.Windows.Forms.Control.WndProc%2A> method corresponds exactly to the Windows `WindowProc` function. For more information about processing Windows messages, see the `WindowProc` function documentation in the Windows Platform SDK reference located in the MSDN library at https://docs.microsoft.com/en-us/.
```
Comment (mairaw) [inline xml/System.Windows.Forms/ToolStrip.xml:4673]:
```suggestion
 The <xref:System.Windows.Forms.Control.WndProc%2A> method corresponds exactly to the Windows `WindowProc` function. For more information about processing Windows messages, see the [WindowProc function](https://docs.microsoft.com/previous-versions/windows/desktop/legacy/ms633573(v=vs.85)).
```

File: xml/System.Windows.Forms/ToolStrip.xml [hunk 33]
```diff
@@ -4670,10 +4670,10 @@
...
+ The following code example demonstrates overriding the <xref:System.Windows.Forms.Control.WndProc%2A> method to handle operating system messages identified in the <xref:System.Windows.Forms.Message> structure. The WM_ACTIVATEAPP operating system message is handled in this example to know when another application is becoming active. For information about the available <xref:System.Windows.Forms.Message.Msg%2A?displayProperty=nameWithType>, <xref:System.Windows.Forms.Message.LParam%2A?displayProperty=nameWithType>, and <xref:System.Windows.Forms.Message.WParam%2A?displayProperty=nameWithType> values, see the Platform SDK documentation reference located in the MSDN library at https://docs.microsoft.com/en-us/. Actual constant values can be found in the windows.h header file included in the Platform SDK (Core SDK section) download, which is also available on MSDN.
```
Comment (mairaw) [inline xml/System.Windows.Forms/ToolStrip.xml:4676]:
```suggestion
 The following code example demonstrates overriding the <xref:System.Windows.Forms.Control.WndProc%2A> method to handle operating system messages identified in the <xref:System.Windows.Forms.Message> structure. The WM_ACTIVATEAPP operating system message is handled in this example to know when another application is becoming active. For information about the available <xref:System.Windows.Forms.Message.Msg%2A?displayProperty=nameWithType>, <xref:System.Windows.Forms.Message.LParam%2A?displayProperty=nameWithType>, and <xref:System.Windows.Forms.Message.WParam%2A?displayProperty=nameWithType> values, see the Platform SDK documentation reference located at <https://docs.microsoft.com>. Actual constant values can be found in the windows.h header file included in the Platform SDK (Core SDK section) download, which is also available on Docs.  
```

File: xml/System.Windows.Forms/HtmlDocument.xml [hunk 34]
```diff
@@ -1929,7 +1927,7 @@ window.alert("Name is " + name + "; address is " + address);
  ]]></format>
         </remarks>
         <altmember cref="M:System.Windows.Forms.HtmlDocument.OpenNew(System.Boolean)" />
-        <related type="Article" href="https://msdn.microsoft.com/library/default.asp?url=/workshop/browser/mshtml/reference/ifaces/document2/write.asp">write Method</related>
+        <related type="Article" href="https://docs.microsoft.com/en-us/">write Method</related>
```
Comment (mairaw) [inline xml/System.Windows.Forms/HtmlDocument.xml:1930]:
```suggestion
```

**Pull request merged**